File: app/src/main/java/com/futsch1/medtimer/helpers/DeleteHelper.java
package com.futsch1.medtimer.helpers;

import android.app.AlertDialog;
import android.content.Context;

import com.futsch1.medtimer.R;

public class DeleteHelper {
    private final Context context;

    public DeleteHelper(Context context) {
        this.context = context;
    }

    public void deleteItem(int messageStringId, ButtonCallback yesClicked, ButtonCallback noClicked) {
        AlertDialog.Builder builder = new AlertDialog.Builder(context);
        builder.setTitle(R.string.confirm);
        builder.setMessage(messageStringId);
        builder.setCancelable(false);
        builder.setPositiveButton(R.string.yes, (dialogInterface, i) -> yesClicked.onButtonClick());
        builder.setNegativeButton(R.string.cancel, (dialogInterface, i) -> noClicked.onButtonClick());
        builder.show();
    }

    public interface ButtonCallback {
        void onButtonClick();
    }
}

File: app/src/main/java/com/futsch1/medtimer/helpers/DeleteHelper.kt
package com.futsch1.medtimer.helpers

import android.content.Context
import com.futsch1.medtimer.R
import com.google.android.material.dialog.MaterialAlertDialogBuilder

object DeleteHelper {
    fun deleteItem(context: Context, messageStringId: Int, yesClicked: () -> Unit, noClicked: () -> Unit) {
        val builder = MaterialAlertDialogBuilder(context)
        builder.setTitle(R.string.confirm)
        builder.setMessage(messageStringId)
        builder.setPositiveButton(R.string.yes) { _, _ -> yesClicked() }
        builder.setNegativeButton(R.string.cancel) { _, _ -> noClicked() }
        builder.setOnCancelListener { _ -> noClicked() }
        builder.show()
    }
}
